src/components/AgencySearch.js

```diff
@@ -17,7 +17,10 @@ class AgencySearch extends Component {
     if (initialShowResults !== this.props.initialShowResults) {
       newState.showResults = initialShowResults
     }
-    if (agency === '') newState.search = ''
+    if (agency === '') {
```

Comment:
i believe multiple setStates within this lifecycle method are batched, so if you wanted, i think you could also write this as below (if you prefer):

``` 
  componentWillReceiveProps({ agency, initialShowResults }) {
    if (initialShowResults !== this.props.initialShowResults) {
      this.setState({ showResults: initialShowResults })
    }
    if (agency === '') {
      this.setState({ search: '', hasSelection: false })
   }
  }
```